st2client/st2client/commands/action_alias.py

```diff
@@ -40,3 +54,106 @@ class ActionAliasGetCommand(resource.ContentPackResourceGetCommand):
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        aliases = self.manager.get_all(**kwargs)
+        matches = match_command_to_alias(args.match_text, aliases)
+        return [match[0] for match in matches]  # show only alias objects
+
+    def run_and_print(self, args, **kwargs):
+        instances = self.run(args, **kwargs)
+        self.print_output(instances, table.MultiColumnTable,
+                          attributes=args.attr, widths=args.width,
+                          json=args.json, yaml=args.yaml)
+
+
+class ActionAliasExecuteCommand(ActionRunCommandMixin,
+                                resource.ResourceCommand):
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionAliasExecuteCommand, self).__init__(
+            resource, 'execute',
+            'Execute the command text by finding a matching ActionAlias.',
+            *args, **kwargs)
+
+        self.parser.add_argument('command_text',
+                                 metavar='command',
+                                 help=help)
+        self.parser.add_argument('-h', '--help',
+                                 action='store_true', dest='help',
+                                 help='Print usage for the given action.')
+        self.parser.add_argument('--trace-tag', '--trace_tag',
+                                 help='A trace tag string to track execution later.',
+                                 dest='trace_tag', required=False)
+        self.parser.add_argument('--trace-id',
+                                 help='Existing trace id for this execution.',
+                                 dest='trace_id', required=False)
+        self.parser.add_argument('-a', '--async',
+                                 action='store_true', dest='async',
+                                 help='Do not wait for action to finish.')
+        self.parser.add_argument('-u', '--user', type=str, default=None,
+                                 help='User under which to run the action (admins only).')
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        aliases = self.manager.get_all(**kwargs)
+        matches = match_command_to_alias(args.command_text, aliases)
+        if len(matches) > 1:
+            raise ActionAliasAmbiguityException("Too many matches for provided command",
+                                                matches=matches)
+        elif len(matches) == 0:
+            raise ActionAliasAmbiguityException("Could not locate an ActionAlias with a "
+                                                "matching format command", matches=matches)
+        match = matches[0]
+        action_alias_db = match[0]
+
+        if not action_alias_db.enabled:
```

Comment:
the tracing logic is different, I looked at what was done in the action.py execute command module and refactored for this scenario. Agreed it's not ideal, the alias execute API doesn't wait for the execution to be completed from what I saw.